src/synth/fluid_synth.c

```diff
@@ -3899,17 +3897,15 @@ fluid_synth_set_reverb_on(fluid_synth_t* synth, int on)
 int
 fluid_synth_set_reverb_preset(fluid_synth_t* synth, int num)
 {
-  int i = 0;
-  while (revmodel_preset[i].name != NULL) {
-    if (i == num) {
-      fluid_synth_set_reverb (synth, revmodel_preset[i].roomsize,
-                              revmodel_preset[i].damp, revmodel_preset[i].width,
-                              revmodel_preset[i].level);
-      return FLUID_OK;
-    }
-    i++;
-  }
-  return FLUID_FAILED;
+  fluid_return_val_if_fail (
+    num < 0 || num >= (int)FLUID_N_ELEMENTS(revmodel_preset),
```

Comment:
Should be `num>=0 && num < FLUID_N_ELEMENTS()` because if this fails (evaluates to false) it returns. But you could also change num back to unsigned, sorry for the confusion.

Reply:
Ouch, you are right, I misunderstood the meaning of the macro, sorry.